D1371R3.md

```diff
@@ -385,29 +384,28 @@ Within the parentheses, `inspect` is equivalent to `switch` and `if` statements
 except that no conversion nor promotion takes place in evaluating the value
 of its condition.
 
-There are two clause variations types inside the `inspect` body:
-
-  - `=>` denotes expression yielding a value.
-  - `:` never yields a value, but instead denotes a statement that is executed
-    in the enclosing context. e.g, a `return` statement will return from the
-    function `inspect` is currently running in.
-
-`inspect` is an expression if it's encountered in an expression-only context,
-`trailing-return-type` is specified, or any `=>` clauses are present.
-It's a statement otherwise.
+`inspect` is an expression in all contexts. Depending on the enclosed patterns it
+may either yield a `void` result or a value, the type of which will be statically
+deduced from the trailing return type or the patterns themselves. This deduction
+is analogous to that performed when determining the return type of a lambda
+expression. An `inspect` expression yields a `void` result if one or more of
```

Comment:
From the previous and next sentence it kind of follows the logic that *all* patterns must yield the same type. So not sure if "one or more" is correct here.

Reply:
I'll see if I can tighten this up